In [3]:
## Dia 1 Carga y exploración de datos

# Instalar la librería de UCI
!pip install ucimlrepo

In [5]:
# Importar librerías necesarias
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from ucimlrepo import fetch_ucirepo

In [7]:
# Cargar el dataset de UCI
auto_mpg = fetch_ucirepo(id=9)

df= pd.concat([auto_mpg.data.features, auto_mpg.data.targets], axis=1)

In [8]:
# Exploración básica
print(df.shape)        # (398, 9)
print(df.dtypes)
print(df.isnull().sum())  # Verás 6 nulls en horsepower
df.describe()

(398, 8)
displacement    float64
cylinders         int64
horsepower      float64
weight            int64
acceleration    float64
model_year        int64
origin            int64
mpg             float64
dtype: object
displacement    0
cylinders       0
horsepower      6
weight          0
acceleration    0
model_year      0
origin          0
mpg             0
dtype: int64


,displacement,cylinders,horsepower,weight,acceleration,model_year,origin,mpg
count,398.000000,398.000000,392.000000,398.000000,398.000000,398.000000,398.000000,398.000000
mean,193.425879,5.454774,104.469388,2970.424623,15.568090,76.010050,1.572864,23.514573
std,104.269838,1.701004,38.491160,846.841774,2.757689,3.697627,0.802055,7.815984
min,68.000000,3.000000,46.000000,1613.000000,8.000000,70.000000,1.000000,9.000000
25%,104.250000,4.000000,75.000000,2223.750000,13.825000,73.000000,1.000000,17.500000
50%,148.500000,4.000000,93.500000,2803.500000,15.500000,76.000000,1.000000,23.000000
75%,262.000000,8.000000,126.000000,3608.000000,17.175000,79.000000,2.000000,29.000000
max,455.000000,8.000000,230.000000,5140.000000,24.800000,82.000000,3.000000,46.600000


## El dataset está cargado.
# Lo que podemos ver:
-398 filas en casi todas las columnas, excepto "horsepower" que tiene 392 (Es decir, que tiene 6 datos nulos)
- "Model Year" tiene el mínimo en 70 (Entiendo que 1970), y el máximo 82 (1982)
- "MPG" significa "Millas por galon de gasolina", se refiere al consumo del coche, que su media está en 23.51
- El objetivo se centra en poder predecir el consumo de un coche a partir de sus características